### PR TITLE
disable autocorrect on NoteComposer when running on macOS #1460

### PR DESCRIPTION
**Pull request by bryanmontz**
## Issues covered
#1460 

## Description
Disables autocorrect on the `NoteComposer` view when running on macOS because it was interfering with composing a note.

## How to test
Running the app on macOS:
1. Navigate to the note composer
2. Begin typing a note
Prior to this change, you would frequently see that macOS would try to autocorrect too quickly and you would get tripped up and write nonsense. After the change, autocorrect is disabled (only on macOS) and the issue doesn't happen anymore.

## Screenshots/Video
On macOS:
Before
before 2024-09-25 07_14_51

After
after 2024-09-25 07_15_42

Regression check: autocorrect still works as expected on iPad:
ipad

### CODE REVIEW

File: Nos/Views/NoteComposer/NoteUITextViewRepresentable.swift [hunk 1]
```diff
@@ -47,6 +47,10 @@ extension NoteTextEditor {
                 .foregroundColor: UIColor.primaryTxt
             ]
             
+            if ProcessInfo.processInfo.isiOSAppOnMac {
```
Comment (mplorentz) [inline Nos/Views/NoteComposer/NoteUITextViewRepresentable.swift:50]: HOW DID YOU FIND THIS API? I HAVE GOOGLED FOR THIS SO MANY TIMES. I AM ANGRY AND ALSO VERY EXCITED.

**Review by mplorentz — APPROVED** [review]

**Comment by mplorentz** [comment]
@bryanmontz are you going to keep searching for a workaround that would let us re-enable autocorrect on Mac?

File: Nos/Views/NoteComposer/NoteUITextViewRepresentable.swift (hunk 1 above)
Comment (joshuatbrown) [reply Nos/Views/NoteComposer/NoteUITextViewRepresentable.swift:50]: Whoa, what??? I've never found this, either! I thought there was no way to detect this! 😲🎉‼️

Comment (bryanmontz) [reply Nos/Views/NoteComposer/NoteUITextViewRepresentable.swift:50]: Haha sooo much googling. I thought for sure there would be a compiler directive, but it makes sense you have to check at runtime I guess.

**Comment by bryanmontz** [comment]
> @bryanmontz are you going to keep searching for a workaround that would let us re-enable autocorrect on Mac?

I wasn't planning on it because:
1. it wasn't part of Linda's request, and
2. it feels like an OS bug and something we can't configure or change.

**Comment by mplorentz** [comment]
@bryanmontz ok, let's just make sure it's clear to Linda that autocomplete will be off on the mac for the foreseeable future. We had turned autocomplete off (on all platforms) to fix this in the past and we had complaints from users about that so I want to make sure she is comfortable with that.

But I also spent time trying to fix it last year it does just seem like an OS bug.

**Comment by mplorentz** [comment]
Oh, I see now that Linda specifically asked about disabling the autocomplete on Mac. Disregard me 👍

**Pull request merged**